Penteract/Assets/Scripts/RangedAI.cpp

```diff
@@ -251,13 +280,29 @@ void RangedAI::EnterState(AIState newState) {
 			}
 		}
 		break;
+	case AIState::STUNNED:
+		if (shot) {
+			animation->SendTriggerSecondary("ShootBeginStun");
```

Comment:
Animation not checked. Consider adding the check of animation at the beginning of the function to avoid thousands of if(animation). All the new cases are missing the nullity check. In case you add the condition at the top of the function, you can remove it from the others cases like AIState::RUN